samples/api/hlsl_shaders/hlsl_shaders.cpp

```diff
@@ -0,0 +1,513 @@
+/* Copyright (c) 2021, Sascha Willems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Using HLSL shaders in Vulkan with the glslang library
+ */
+
+#include "hlsl_shaders.h"
+
+VKBP_DISABLE_WARNINGS()
+#include <SPIRV/GLSL.std.450.h>
+#include <SPIRV/GlslangToSpv.h>
+#include <StandAlone/ResourceLimits.h>
+#include <glslang/Include/ShHandle.h>
+#include <glslang/OSDependent/osinclude.h>
+VKBP_ENABLE_WARNINGS()
+
+VkPipelineShaderStageCreateInfo HlslShaders::load_hlsl_shader(const std::string &file, VkShaderStageFlagBits stage)
+{
+	std::vector<uint32_t> spirv;
+	std::string           info_log;
+
+	// Compile HLSL to SPIR-V
+
+	// Initialize glslang library
+	glslang::InitializeProcess();
+
+	EShMessages messages = static_cast<EShMessages>(EShMsgReadHlsl | EShMsgDefault | EShMsgVulkanRules | EShMsgSpvRules);
+
+	EShLanguage language{};
+	switch (stage)
+	{
+		case VK_SHADER_STAGE_VERTEX_BIT:
+			language = EShLangVertex;
+			break;
+		case VK_SHADER_STAGE_FRAGMENT_BIT:
+			language = EShLangFragment;
+			break;
+	}
```

Comment:
add default case

Reply:
default is helpful here.

Reply:
default could be an `assert(false && "VkShaderStageFlagBits not supported")`?